docs/architecture/autofill/collecting-page-details.md

```diff
@@ -0,0 +1,243 @@
+
+#### Manifest v2
+
+In a browser extension running Manifest v2, the `commands.background.ts` background page is
+listening for the `autofill_login` command. This background page executes the
+`collectPageDetailsForContentScript()` method on `main.background.js`, which broadcasts the
+`collectPageDetails` message to the `autofill.js` content script.
+
+After generating the page details, the `autofill.js` content script broadcasts a
+`collectPageDetailsResponse` message with an `autofill_cmd` sender. The `runtime.background.js`
+background page is listening for this message and receives it.
+
+```kroki type=plantuml
+@startuml
+
+actor User
+
+box "Content Scripts" #B8B8B8
+participant autofill.js as autofill
+end box
+
+box "Background Pages" #E5E8E8
+participant commands.background.ts as commands
+participant runtime.background.ts as runtime
+participant main.background.ts as main
+end box
+
+User -> commands : autofill_login
+commands --> main : collectPageDetailsForContentScript()
+main -> autofill : collectPageDetails [sender: autofill_cmd]
+autofill -> runtime : collectPageDetailsResponse [sender: autofill_cmd]
+
+@enduml
+```
+
+#### Manifest v3
+
+For browser extensions running Manifest v3, the background pages are replaced with the
+`commandListener`. The `commandListener` is listening for the `autofill_login` command, and it
+responds by broadcasting the `collectPageDetailsImmediately` command.
+
+The `collectPageDetailsImmediately` is different from the `collectPageDetails` command, in that the
+response is **not** another message broadcast through the browser command API. Instead, the
+`autofill.js` content script performs the page details generation and returns the response
+asynchronously through a Promise.
+
+```kroki type=plantuml
+@startuml
+
+actor User
+
+box "Content Scripts" #B8B8B8
+participant autofill.js as autofill
+end box
+
+
+box "Listeners" #7A7A7A
+participant commandListener as listener
+end box
+
+User -> listener : autofill_login
+listener -> autofill : collectPageDetailsImmediately
+autofill --> listener
+@enduml
+```
+
+### Requesting page details from the extension UI
+
+There are two ways that a user can request an autofill from the Bitwarden browser extension UI:
+
+- Clicking the Autofill button when viewing an item in their vault (`view.component.ts`), or
+- Clicking on a vault item on the Current Tab view (`current-tab.component.ts`)
+
+In both of these cases, the component issues a `collectPageDetails` command with the extension
+instance's unique `BroadcasterSubscriptionId` as the `sender`. The `autofill.js` content script
+generates the page details and responds with a `collectPageDetailsResponse` message with the same
+`sender`, ensuring that the message is received properly by the correct sender.
+
+```kroki type=plantuml
+@startuml
+actor User
+
+box "Content Scripts" #B8B8B8
+participant autofill.js as autofill
+end box
+
+box "Extension UI" #1252A3
```

Comment:
I find the contrast on this color pretty harsh against the white background and the black foreground. I think it could just be a pale grey like the Content Scripts box.

Reply:
I apologize - I thought I changed it but I didn't commit the change.